app/v2/notifications/post_notifications.py

```diff
@@ -393,10 +393,13 @@ def process_precompiled_letter_notifications(*, letter_data, api_key, template,
 def get_reply_to_text(notification_type, form, template):
     reply_to = None
     if notification_type == EMAIL_TYPE:
-        service_email_reply_to_id = form.get("email_reply_to_id", None)
-        reply_to = check_service_email_reply_to_id(
-            str(authenticated_service.id), service_email_reply_to_id, notification_type
-        ) or template.get_reply_to_text()
+        if template.reply_to_email is not None:
+            reply_to = template.reply_to_email
+        else:
+            service_email_reply_to_id = form.get("email_reply_to_id", None)
+            reply_to = check_service_email_reply_to_id(
+                str(authenticated_service.id), service_email_reply_to_id, notification_type
+            ) or template.get_reply_to_text()
```

Comment:
The logic looks fine, but I recommend changes to the original code, which you haven't altered.

I've seen this use of "or" in JavaScript, but I haven't seen it in Python before.  check_service_replay_to_id returns a string or None, and this block is treating None like a boolean.  bool(None)=False, but None and False are not equivalent.  Treating them as equivalent can cause problems.  I recommend this code instead:

```
service_email_reply_to_id = form.get("email_reply_to_id", None)
if service_email_reply_to_id is not None:
    reply_to = check_service_email_replay_to_id(...)  # str or None
if service_email_reply_to_id is None:
    reply_to = template.get_reply_to_text()
```

The function check_service_email_replay_to_id, in app/notifications.py, line 184, also treats None like a boolean.  I would take this opportunity to change that to "if reply_to_id is not None:".  Better yet, raise ValueError if the function is passed None to avoid what I call a "do something maybe" function.

Reply:
Yep, I'm ok with these changes and will get them pushed up soon. I did start working on a test and I ended up breaking something so as soon as I get that fixed I will have all these changes pushed up at once.

Reply:
I was able to add in a test that checked the `reply_to_text` in the notification which should be where this email is now going to be set. I also updated the code per the suggestion above which was a quick insert and a clean change.